api/maistra/v1alpha1/session_types_test.go

```diff
@@ -0,0 +1,73 @@
+package v1alpha1_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/maistra/istio-workspace/api/maistra/v1alpha1"
+)
+
+var _ = Describe("API manipulation", func() {
+
+	Context("when flipping readiness", func() {
+
+		var components v1alpha1.StatusComponents
+		compoentOne := "one"
+		compoentTwo := "two"
+
+		BeforeEach(func() {
+			components = v1alpha1.StatusComponents{}
+		})
+
+		It("should set as pending", func() {
+			components.SetPending(compoentOne)
+
+			Expect(components.Pending).To(HaveLen(1))
+		})
+		It("should set as Ready", func() {
+			components.SetReady(compoentOne)
+
+			Expect(components.Ready).To(HaveLen(1))
+		})
+		It("should set as UnReady", func() {
+			components.SetUnReady(compoentOne)
+
+			Expect(components.UnReady).To(HaveLen(1))
+		})
+
+		It("should flip from pending to ready", func() {
+			components.SetPending(compoentOne)
+			Expect(components.Pending).To(HaveLen(1))
```

Comment:
Not sure we really need this kind of assertions. Adds noise perhaps? It's a test fixture after all.

Reply:
What do you mean this kind of assertion? What's a test fixture?

Reply:
Test fixture - initial state of things on which you perform some logic and validate the outcome. In this case, you have `componentOne` in pending state. I don't think that should be validate. It's anyway tested earlier in dedicated test.

Reply:
Ah, yea, that's true.

Reply:
fixed https://github.com/maistra/istio-workspace/pull/875/commits/06b6ac92b8d0d065c6477ae263d5b91390708479